gobblin-data-management/src/main/java/gobblin/data/management/conversion/hive/converter/HiveAvroToOrcConverter.java

```diff
@@ -0,0 +1,168 @@
+        throw new IllegalArgumentException("Both partitions info and partions must be present, if one is specified");
+      }
+      Map<String, String> partitionDDLInfo = new HashMap<>();
+      Map<String, String> partitionDMLInfo = new HashMap<>();
+      List<String> pInfo = Splitter.on(",").omitEmptyStrings().trimResults().splitToList(partitionsInfoString);
+      List<String> pType = Splitter.on(",").omitEmptyStrings().trimResults().splitToList(partitionsTypeString);
+      if (pInfo.size() != pType.size()) {
+        throw new IllegalArgumentException("partitions info and partitions type list should of same size");
+      }
+      for (int i=0; i<pInfo.size(); i++) {
+        List<String> partitionInfoParts = Splitter.on("=").omitEmptyStrings().trimResults().splitToList(pInfo.get(i));
+        String partitionType = pType.get(i);
+        if (partitionInfoParts.size() != 2) {
+          throw new IllegalArgumentException(
+              String.format("Partition details should be of the format partitionName=partitionValue. Recieved: %s",
+                  pInfo.get(i)));
+        }
+        partitionDDLInfo.put(partitionInfoParts.get(0), partitionType);
+        partitionDMLInfo.put(partitionInfoParts.get(0), partitionInfoParts.get(1));
+      }
+      if (partitionDDLInfo.size() > 0) {
+        optionalPartitionsDDLInfo = Optional.of(partitionDDLInfo);
+      }
+      if (partitionDMLInfo.size() > 0) {
+        optionalPartitionsDMLInfo = Optional.of(partitionDMLInfo);
+      }
+    }
+
+    // Create DDL statement
+    String createFlattenedTableDDL = HiveAvroORCQueryUtils
+        .generateCreateTableDDL(flattenedSchema,
+            orcTableName,
+            orcTableLocation,
+            Optional.of(conversionEntity.getHiveUnit().getDbName()),
+            optionalPartitionsDDLInfo,
+            Optional.<List<String>>absent(),
+            Optional.<Map<String, HiveAvroORCQueryUtils.COLUMN_SORT_ORDER>>absent(),
+            Optional.<Integer>absent(),
+            Optional.<String>absent(),
+            Optional.<String>absent(),
+            Optional.<String>absent(),
+            Optional.<Map<String, String>>absent());
+    conversionEntity.appendQuery(createFlattenedTableDDL);
+    log.info("Create DDL: " + createFlattenedTableDDL);
+
+    // Create DML statement
+    String insertInORCTableDML = HiveAvroORCQueryUtils
+        .generateTableMappingDML(outputSchema, flattenedSchema, avroTableName, orcTableName,
+            Optional.of(conversionEntity.getHiveUnit().getDbName()),
+            Optional.of(conversionEntity.getHiveUnit().getDbName()),
+            optionalPartitionsDMLInfo,
+            Optional.<Boolean>absent(), Optional.<Boolean>absent());
+    conversionEntity.appendQuery(insertInORCTableDML);
+    log.info("Conversion DML: " + insertInORCTableDML);
+
+    log.info("Conversion Query " + conversionEntity.getConversionQuery());
```

Comment:
I think submitting a DDL creation event would be useful. I suggest we tag all events with dataset metadata like tablename, dbname, partition etc. It would be easy to generate reports on them